### ISSUE TITLE
FakeFS::FileSystem.clone on directory with '+' character

### ISSUE THREAD

**Issue by panozzaj** [fakefs/fakefs#133]
I cannot FakeFS::FileSystem.clone('test/a+b/something'), but I can FakeFS::FileSystem.clone('test/aplusb/something') if those files are present on the current filesystem. When the plus is present, I get

    Errno::ENOENT:
       No such file or directory - No such file or directory - /fullpath/test/a+b/something/otherthings.txt

otherwise it seems to work fine.

**Comment by sometimesfood** [comment]
Apparently these files are created in the fake filesystem, but cannot be found by `FileSystem::find`.

In the example below, the first assert passes, but the second fails.
```ruby
  def test_can_create_directories_with_plus_in_name                                                                                                          
    path = '/foo/bar+baz'                                                                                                                                    
    FileUtils.mkdir_p(path)                                                                                                                                  
    assert_kind_of FakeFS::FakeDir, FakeFS::FileSystem.fs['foo']['bar+baz']                                                                                  
    assert_kind_of FakeFS::FakeDir, FakeFSS::FileSystem.find(path)                                                                                           
  end
```

**Comment by sometimesfood** [comment]
Could this be related to #264?

**Comment by sometimesfood** [comment]
This is fixed in fakefs 0.9. Commit 023fca7fe92d5ccfd48c9044b36493450e0151b5 seems to have helped.

@panozzaj: Can you please confirm and close this issue?

**Comment by panozzaj** [comment]
@sometimesfood Testing using the [test you provided](https://github.com/fakefs/fakefs/issues/133#issuecomment-102586707) and [the test I provided](https://github.com/fakefs/fakefs/pull/134/commits/3152e70f3345c162d70ec91fa3b0f7812c80d32e) both give success for fakefs `~> 0.9.0`, and fails when using `< 0.9.0`. So I believe this is fixed! 👏
